src/applications/mhv-secure-messaging/tests/e2e/secure-messaging-pagination.cypress.spec.js

```diff
@@ -114,4 +115,52 @@ describe('Secure Messaging Reply', () => {
       cy.get(el).should('be.visible');
     });
   });
+
+  it('verify correct number of messages rendered after filtering reduces page count', () => {
+    const initialThreadLength = 28;
+    const filteredThreadLength = 5;
+
+    mockMessagesPageOne.data.forEach(item => {
+      const currentItem = item;
+      currentItem.attributes.threadPageSize = initialThreadLength;
+    });
+    mockMessagesPageTwo.data.forEach(item => {
+      const currentItem = item;
+      currentItem.attributes.threadPageSize = initialThreadLength;
+    });
+
+    PatientInboxPage.loadInboxMessages(mockMessagesPageOne);
+    cy.get('va-pagination').should('be.visible');
+
+    SecureMessagingSite.clickAndLoadVAPaginationNextMessagesButton(
+      2,
+      mockMessagesPageTwo,
+    );
+    SecureMessagingSite.verifyPaginationMessagesDisplayedText(
+      11,
+      20,
+      initialThreadLength,
+    );
+
+    const filterResultResponse = PatientFilterPage.createCategoryFilterMockResponse(
+      filteredThreadLength,
+      'Appointment',
+      mockMessagesPageOne,
+    );
+
+    PatientFilterPage.openAdditionalFilter();
+    PatientFilterPage.selectAdvancedSearchCategory('Appointment');
+    PatientFilterPage.clickApplyFilterButton(filterResultResponse);
+
+    cy.get('va-pagination').should('not.exist');
+    PatientFilterPage.verifyFilterResponseLength(filterResultResponse);
+
+    cy.contains(
+      `Showing 1 to ${filteredThreadLength} of ${filteredThreadLength}`,
+    );
+    cy.contains('End of search results');
+
+    cy.injectAxe();
+    cy.axeCheck(AXE_CONTEXT);
```

Comment:
both options are OK, this is just a shorthand function
```suggestion
    cy.injectAxeThenAxeCheck(AXE_CONTEXT);
```